code/datums/components/cult_held_body.dm

```diff
@@ -0,0 +1,67 @@
+/**
+ * A component for tracking and manipulating bodies held inside constructs/shades
+ * Will drop the body/brain when the parent dies or is deleted.
+ */
+/datum/component/construct_held_body
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	/// A reference to either a mob, or a brain organ that is held inside our parent. Will drop when parent dies/is deleted
+	var/atom/movable/held_body
+
+/datum/component/construct_held_body/Initialize(atom/movable/new_body)
+	. = ..()
+	add_body(new_body)
+
+/datum/component/construct_held_body/PostTransfer()
+	held_body.forceMove(parent) // forcemove them to the new parent
+
+/datum/component/construct_held_body/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_DEATH, PROC_REF(drop_body))
+	RegisterSignal(parent, COMSIG_PARENT_QDELETING, PROC_REF(drop_body))
+	RegisterSignal(parent, COMSIG_SHADE_TO_CONSTRUCT_TRANSFER, PROC_REF(transfer_held_body))
+
+/datum/component/construct_held_body/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_DEATH)
+	UnregisterSignal(parent, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(parent, COMSIG_SHADE_TO_CONSTRUCT_TRANSFER)
```

Comment:
You can add multiple signals to `UnregisterSignal`, up to you

Reply:
nope

Reply:
👍